src/Data/Variant/Gen.purs

```diff
@@ -0,0 +1,176 @@
+module Data.Variant.Gen where
+
+import Prelude
+
+import Control.Monad.Gen (class MonadGen, frequency, oneOf)
+import Data.List as L
+import Data.NonEmpty (NonEmpty, (:|))
+import Data.Symbol (class IsSymbol)
+import Data.Tuple (Tuple(..))
+import Data.Variant (Variant, inj)
+import Prim.Row as Row
+import Prim.RowList as RL
+import Record as Record
+import Type.Proxy (Proxy(..))
+
+-- Generate a `Variant` with uniform probability given a record whose
+-- generators' labels correspond to the `Variant`'s labels
+--
+-- ```purs
+-- genMyVariant :: Gen (Variant (id :: String, unit :: Unit))
+-- genMyVariant =
+--   genVariantUniform
+--     { id: genAlphaString
+--     , unit: pure unit
+--     }
+-- ```
+--
+-- Inspired by https://github.com/JordanMartinez/purescript-veither/blob/9fb195b7bd97cfcf21bbd389b55002ca458f1bb4/src/Data/Veither.purs#L461
```

Comment:
What is the inspiration exactly and how is it relevant? Do we need to fulfill a license? If so, I don't think the link is enough. If there are important, non-obvious considerations, it probably warrants a general explanation comment somewhere documenting the cases with this link as a reference. Otherwise, I don't think the link is necessary since it's not obvious that it's contextually relevant to anything.